sdk/storage/azure-storage-queue/azure/storage/queue/_models.py

```diff
@@ -275,9 +277,20 @@ def __init__(self, command, results_per_page=None, continuation_token=None):
         )
         self._command = command
         self.results_per_page = results_per_page
+        self.messages_to_retrieve = messages_to_retrieve
 
     def _get_next_cb(self, continuation_token):
         try:
+            if self.messages_to_retrieve is not None:
+                if self.results_per_page is None:
+                    self.results_per_page = 1
```

Comment:
I think maybe we should have this be the default value when popping the `kwarg` rather than checking for it here?

Reply:
I am definitely open to suggestions to this- I think my thinking here was that I didn't want to change the default value from `None`, but in testing if we don't provide a value for `results_per_page` but we do provide a value for `mesages_to_retrieve` (thus falling into this conditional), we will face a `TypeError` for using an inequality on `None` (line 289)

Reply:
Resolved in offline discussion.